### PR TITLE
[ruleporn] Add support for non Nuevo urls, add tests, strip title

### PR DESCRIPTION
**Pull request by savicki**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x ] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x ] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x ] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x ] Bug fix
- [x ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Add support for non nuevo urls and tests for these videos

### CODE REVIEW

File: youtube_dl/extractor/ruleporn.py [hunk 1]
```diff
@@ -16,29 +17,93 @@ class RulePornIE(NuevoBaseIE):
...
-        info.update({
-            'display_id': display_id,
-            'title': title,
-            'description': description,
-            'age_limit': 18
-        })
+        video_url = self._search_regex(
+            r'<div\s+id=\"player\">\s+<iframe[\w\W]+?(?:src=[\"\']?(?P<url>[^\"\' ]+))', webpage, 'video url')
+
+        mobj = re.search(r'lovehomeporn\.com/embed/(\d+)', video_url)
```
Comment (dstftw) [inline youtube_dl/extractor/ruleporn.py]: `_search_regex` with `default=None` instead.

File: youtube_dl/extractor/ruleporn.py [hunk 2]
```diff
@@ -16,29 +17,93 @@ class RulePornIE(NuevoBaseIE):
...
+                'description': description,
+                'age_limit': 18
+            }) 
+
+        else:
+
+            video_page = self._download_webpage(video_url, display_id)
+
+            js_str = self._search_regex(
+                r'<script\s+type=\"text/javascript\">\s+var\s+item\s+=\s+({[^}]+});', video_page, 'js_str')
```
Comment (dstftw) [inline youtube_dl/extractor/ruleporn.py]: No need to include script tag. It also won't work if JSON happen to contain inner dict.

File: youtube_dl/extractor/ruleporn.py [hunk 3]
```diff
@@ -16,29 +17,93 @@ class RulePornIE(NuevoBaseIE):
...
+
+            js_obj = self._parse_json(js_str, display_id)
+            
+            video_id = '%s' % js_obj.get("id")
```
Comment (dstftw) [inline youtube_dl/extractor/ruleporn.py]: Single quotes.

File: youtube_dl/extractor/ruleporn.py [hunk 4]
```diff
@@ -16,29 +17,93 @@ class RulePornIE(NuevoBaseIE):
...
+            duration = js_obj.get("video_duration")
```
Comment (dstftw) [inline youtube_dl/extractor/ruleporn.py]: `int_or_none`.

File: youtube_dl/extractor/ruleporn.py (hunk 2 above)
Comment (savicki) [inline youtube_dl/extractor/ruleporn.py]: Fixed, not it will

File: youtube_dl/extractor/ruleporn.py (hunk 1 above)
Comment (savicki) [inline youtube_dl/extractor/ruleporn.py]: thank-you.

File: youtube_dl/extractor/funnyordie.py [hunk 5]
```diff
@@ -78,7 +80,7 @@ def _real_extract(self, url):
 
         return {
             'id': video_id,
-            'title': post['name'],
+            'title': post.get('name'),
             'description': post.get('description'),
             'thumbnail': post.get('picture'),
             'formats': formats,
```
Comment (dstftw) [inline youtube_dl/extractor/funnyordie.py]: Unrelated changes.

**Comment by savicki** [comment]
Done!

**Comment by dstftw** [comment]
Check code with flake8.

**Comment by savicki** [comment]
my bad, sorry

**Comment by savicki** [comment]
btw, it's fixed now.. all flake8 warnings

**Comment by dstftw** [comment]
Squash commits.
